lib/misc.py

```diff
@@ -33,6 +33,15 @@ def __call__(self, *args, **kwargs):
 # Argument: command line arguments Namespace. Do not need to call sys.exit()
 # because caller manages that.
 
+def delete(cli):
+   img = ch.Storage(cli.storage)
+   img_ref = ch.Image_Ref()
+   img_ref.name = cli.image_ref
```

Comment:
Here I think you want to let the `Image` constructor do its thing. That is, use `cli.image_ref` to create an `Image` object, and then I think you can ask it where its unpack directory would be.